03-rnaseq/pathway-analysis_rnaseq_03_gsva.Rmd

```diff
@@ -0,0 +1,475 @@
+normalized_counts <- assay(dds_norm) %>%
+  as.data.frame() %>% # Make into a data frame
+  tibble::rownames_to_column("ensembl_id") # Make Gene IDs into their own column
+```
+
+**Review**
+
+### Import Gene Sets
+
+The function that we will use to run GSVA wants the gene sets to be in a list where each pathway has its own gene list inside of it. 
+
+Here we are obtaining the pathway information from the main function of the [`msigdbr`](https://cran.r-project.org/web/packages/msigdbr/index.html) package [@Igor2020].
+Because we are using human data in this example, we supply the formal organism name to the `species` argument. 
+We will want only the hallmark pathways, so we use the `category = "H"` argument.
+
+```{r}
+hallmark_gene_sets <- msigdbr::msigdbr(
+  species = "Homo sapiens", # Can change this to what species you need
+  category = "H" # Only hallmark genes
+)
+```
+
+`hallmark_gene_sets` has which genes are a part of which pathway in a data frame format.
+For this example we will use Entrez IDs (but note that there's gene symbols we could just as easily use).
+The info we need is in two columns: `entrez_gene` contains the gene ID and `gs_name` contains the name of the pathway that the gene is a part of. 
+
+To make this into the list format we need, we can use `split()` function.
+Here we want `entrez_gene` to be made into smaller lists for each pathway.
+
+```{r}
+hallmarks_list <- split(
+  hallmark_gene_sets$entrez_gene, # The genes we want split into lists
+  hallmark_gene_sets$gs_name # The pathways made as the higher levels of the list
+)
+```
+
+What does this `hallmarks_list` look like?
+
+```{r}
+head(hallmarks_list)
+```
+
+Looks like we have a list of gene sets with associated Entrez IDs.
+
+In our gene expression data frame, `expression_df` we have Ensembl gene identifiers.
+So we will need to convert our Ensembl IDs into Entrez IDs for GSVA.
+
+**ALMOST IDENTICAL TO MICROARRAY GSVA** (A few wording rearrangements and `expression_df` -> `normalized_counts`)
+
+### Gene identifier conversion
+
+We're going to convert our identifiers in `expression_df` to Entrez IDs, but you can, with the change of a single argument, use the same code to convert to many other types of identifiers!
+
+The annotation package `org.Hs.eg.db` contains information for different identifiers [@Carlson2020-human].
+`org.Hs.eg.db` is specific to _Homo sapiens_ -- this is what the `Hs` in the package name is referencing.
+
+We can see what types of IDs are available to us in an annotation package with `keytypes()`.
+
+```{r}
+keytypes(org.Hs.eg.db)
+```
+
+We' use this package to convert from Ensembl gene IDs (`ENSEMBL`) to Entrez IDs (`ENTREZID`) -- since this is the IDs we used in our `hallmarks_list` in the previous step. 
+But, we could just as easily use it to convert from an Ensembl transcript ID (`ENSEMBLTRANS`) to gene symbols (`SYMBOL`) if we had built `hallmarks_list` using gene symbols. 
```

Comment:
Why are transcript IDs getting a mention here? The first part is generally true, but not relevant to the `hallmarks_list` as far as I know.